whois-api/src/main/java/net/ripe/db/whois/api/rest/DomainObjectService.java

```diff
@@ -0,0 +1,222 @@
+
+            final List<Update> updates = extractUpdates(resources, credentials);
+
+            final WhoisResources updatedResources = updatePerformer.performUpdates(updateContext, origin, updates, Keyword.NEW, request);
+
+            validateUpdates(updateContext, updates, updatedResources);
+
+            return createResponse(request, updatedResources, Response.Status.OK);
+
+        } catch (WebApplicationException e) {
+            throw specializeWebApplicationException(e, request, resources); // note: response is attached to exception
+
+        } catch (UpdateFailedException e) {
+            return createResponse(request, e.whoisResources, e.status);
+
+        } catch (Exception e) {
+            updatePerformer.logError(e);
+            LOGGER.error("Unexpected", e);
+            return createResponse(request, resources, Response.Status.INTERNAL_SERVER_ERROR);
+
+        } finally {
+            updatePerformer.closeContext();
+        }
+    }
+
+    private WebApplicationException specializeWebApplicationException(
+            final WebApplicationException e,
+            final HttpServletRequest request,
+            final WhoisResources resources) {
+
+        final Response response = e.getResponse();
+
+        switch (response.getStatus()) {
+            case HttpStatus.UNAUTHORIZED_401:
+                return new NotAuthorizedException(createResponse(request, resources, Response.Status.UNAUTHORIZED));
+
+            case HttpStatus.INTERNAL_SERVER_ERROR_500:
+                return new InternalServerErrorException(createResponse(request, resources, Response.Status.INTERNAL_SERVER_ERROR));
+
+            default:
+                return new BadRequestException(createResponse(request, resources, BAD_REQUEST));
+        }
+    }
+
+    private void validateUpdates(UpdateContext updateContext, List<Update> updates, WhoisResources resources) {
+
+        for (Update update : updates) {
+            final UpdateStatus status = updateContext.getStatus(update);
+
+            switch (status) {
+                case SUCCESS: // nothing to do
+                    break;
+
+                case FAILED_AUTHENTICATION:
+                    throw new UpdateFailedException(Response.Status.UNAUTHORIZED, resources);
+
+                case EXCEPTION:
+                    throw new UpdateFailedException(Response.Status.INTERNAL_SERVER_ERROR, resources);
+
+                default:
+                    if (updateContext.getMessages(update).contains(UpdateMessages.newKeywordAndObjectExists())) {
+                        throw new UpdateFailedException(Response.Status.CONFLICT, resources);
+                    } else {
+                        throw new UpdateFailedException(BAD_REQUEST, resources);
+                    }
+            }
+        }
+    }
+
+    private List<Update> extractUpdates(final WhoisResources whoisResources, final Credentials credentials) {
+
+        List<Update> result = Lists.newArrayList();
```

Comment:
No limit to the number of domain objects? Will this create a problem in the update module?